tests/xdg_popup.cpp

```diff
@@ -174,41 +174,42 @@ class XdgPopupManagerBase
         surface.attach_buffer(window_width, window_height);
         surface_rendered = false;
         wl_surface_commit(surface);
-        client.dispatch_until([this]() { return surface_rendered; });
+        client->dispatch_until([this]() { return surface_rendered; });
         the_server.move_surface_to(surface, window_x, window_y);
     }
 
     void map_popup(PositionerParams const& params)
     {
-        popup_surface.emplace(client);
+        popup_surface.emplace(*client);
         setup_popup(params);
         wl_surface_commit(popup_surface.value());
         dispatch_until_popup_configure();
         popup_surface.value().attach_buffer(params.popup_size.first, params.popup_size.second);
         bool surface_rendered{false};
         popup_surface.value().add_frame_callback([&surface_rendered](auto) { surface_rendered = true; });
         wl_surface_commit(popup_surface.value());
-        client.dispatch_until([&surface_rendered]() { return surface_rendered; });
+        client->dispatch_until([&surface_rendered]() { return surface_rendered; });
     }
 
     void unmap_popup()
     {
         clear_popup();
-        popup_surface = std::experimental::nullopt;
+        popup_surface = std::nullopt;
     }
 
-    virtual auto popup_position() const -> std::experimental::optional<std::pair<int, int>> = 0;
+    virtual auto popup_position() const -> std::optional<std::pair<int, int>> = 0;
+    virtual auto child() -> std::unique_ptr<XdgPopupManagerBase> = 0;
```

Comment:
I was confused by this on the first-pass review. This isn't *returning* whatever child the popup has, right? It is creating a new child popup?

If it's going to do that it should have some sort of verb in the name; `create_child_popup()` maybe?